nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -56,7 +56,7 @@ let
     REDIRECT_STATUS   = "200";
   };
 
-  recommendedProxyConfig = pkgs.writeText "nginx-recommended-proxy-headers.conf" ''
+  recommendedProxyConfig = ''
```

Comment:
This should probably be reverted, it keeps the generated text size down.

Reply:
My nginx config is almost 1700 lines long and when researching things this was a constant annoyance because I often would need to open that file to know which options where already set in it. If we would be using include more often for eg vhosts than this would be normal but right now it is rarely used eg mime types, fastcgi params and uwsgi params use it. But for proxy settings half of the settings are in vhosts and the other half in this file which is not optimal.

Reply:
I don't necessarily agree this warrants the removal of that option, my NGINX config is also very long, but it feels okay to me to know those are in a file.
My proposal is to keep the current behavior in this PR so we can merge it and you can open a PR to discuss removing this file?

Reply:
#207140